crates/lsp/src/lib.rs

```diff
@@ -239,7 +239,7 @@ fn url_to_code_description(url: &Option<String>) -> Option<CodeDescription> {
 }
 
 impl<L: LSPLang> Backend<L> {
-  pub fn new(client: Client, rules: RuleCollection<L>) -> Self {
+  pub fn new(client: Client, rules: anyhow::Result<RuleCollection<L>>) -> Self {
```

Comment:
This change will assume lsp must be used with anyhow. However, IMHO LSP should be neutral to error handling.
What about Using std::Result<RuleCollection, String> here?